app/decorators/service_provider_session_decorator.rb

```diff
@@ -39,10 +40,14 @@ def sp_name
   end
 
   def sp_return_url
-    sp.return_to_sp_url
+    if (request_url = session[:request_url])
```

Comment:
I thing this might be clearer if we memoize `session[:request_url]` as another private method named `request_url` and then call `request_url.present?` here 

on first read of this, I thought we were checking for equality rather than assigning a variable.